commons/src/test/java/org/openmrs/eip/BaseCamelTest.java

```diff
@@ -32,24 +27,29 @@
 import org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener;
 import org.springframework.test.context.support.DirtiesContextTestExecutionListener;
 
-import ch.qos.logback.classic.Level;
-import ch.qos.logback.classic.LoggerContext;
-import ch.qos.logback.classic.spi.LoggingEvent;
-import ch.qos.logback.core.read.ListAppender;
+import java.util.List;
+
+import static org.apache.camel.builder.AdviceWith.adviceWith;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.slf4j.Logger.ROOT_LOGGER_NAME;
 
 /**
  * Base class for camel route tests and processors
  */
-@RunWith(CamelSpringRunner.class)
-@SpringBootTest(classes = TestConfig.class)
-@TestExecutionListeners(value = { DirtiesContextBeforeModesTestExecutionListener.class, MockitoTestExecutionListener.class,
-        DependencyInjectionTestExecutionListener.class, DirtiesContextTestExecutionListener.class,
-        ResetMocksTestExecutionListener.class })
+
+@CamelSpringBootTest
+@SpringBootTest(classes = { TestConfig.class })
+@TestExecutionListeners(value = {
+		DirtiesContextBeforeModesTestExecutionListener.class,
+		MockitoTestExecutionListener.class,
+		DependencyInjectionTestExecutionListener.class,
+		DirtiesContextTestExecutionListener.class,
+		ResetMocksTestExecutionListener.class })
 @TestPropertySource(properties = "logging.config=classpath:logback-test.xml")
 @TestPropertySource(properties = "camel.component.direct.block=false")
 @TestPropertySource(properties = "openmrs.eip.log.level=DEBUG")
 @TestPropertySource(properties = "logging.level.org.openmrs.eip=DEBUG")
-@DirtiesContext
+@TestPropertySource(properties = "camel.springboot.routes-collector-enabled=true")
```

Comment:
That's good (I guess), needing `@DirtiesContext` is in general a bad sign, so great that we can get rid of it.